### ISSUE TITLE
Safety section in outer block doc not detected

### ISSUE THREAD

**Issue by MoAlyousef** [rust-lang/rust-clippy#7327]
Hi
Not a blocker, just noticed an issue with outer block docs.

Given the following code:
```rust
#![warn(clippy::missing_safety_doc)]

/**
    does blah
    # Safety
    boom
*/
pub unsafe fn blah() {}    
```

I expected to see this pass that clippy lint, escpecially since the docs reflect appropriately.

Instead, clippy outputs:
```
warning: unsafe function's docs miss `# Safety` section
 --> src/lib.rs:8:1
  |
8 | pub unsafe fn blah() {}    
  | ^^^^^^^^^^^^^^^^^^^^^^^
  |
note: the lint level is defined here
 --> src/lib.rs:1:9
  |
1 | #![warn(clippy::missing_safety_doc)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#missing_safety_doc

warning: 1 warning emitted

    Finished dev [unoptimized + debuginfo] target(s) in 0.53s
```
The problem is that changing the indentation and moving it to the left, fixes the issue, but messes doc rendering of any inlined code in the example section (which I have omitted here).

### Meta

- `cargo clippy -V`:  clippy 0.0.212 (7eac88a 2020-11-16). But also reproducible in the playground with latest clippy (https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=8e76ffaa45b5e46af92b9614f44797ec)
- `rustc -Vv`: 1.48.0

**Comment by giraffate** [comment]
Thanks for the report!

It's a duplicate as https://github.com/rust-lang/rust-clippy/issues/5593.

**Comment by MoAlyousef** [comment]
Oh I’ll close this issue then. 
Thank you
